code/modules/status_system/statusEffects.dm

```diff
@@ -1657,6 +1657,29 @@
 			else
 				. += " You will take toxic damage."
 
+/datum/statusEffect/perfumed
+	id = "perfumed"
+	name = "Perfumed"
+	desc = "You have perfume on you and smell wonderful, you feel extra resistant to miasma and other gross things."
+	icon_state = "perfumed"
+	unique = 1
+	maxDuration = 10 MINUTES
+
+	onUpdate()
+		if(owner.reagents.get_reagent_amount("miasma"))
+			owner.reagents.remove_reagent("miasma", 5)
+			owner.changeStatus("perfumed", -5 SECONDS)
+
+		var/mult = 1
```

Comment:
not much point using `probmult` if you're just setting `mult` to 1, if you want it to be mult adjusted then use the `timePassed` parameter to `onUpdate`

Reply:
Swapped to using prob